devel/release

```diff
@@ -161,7 +161,7 @@ merge-to-release-branch() {
 build-dist() {
     rm -rfv dist augur.egg-info
     python3 setup.py clean
-    python3 setup.py sdist bdist_wheel
+    python3 -m build
```

Comment:
When using `build`, I believe the `python3 setup.py clean` shouldn't be necessary either.

Reply:
Do you have a source for this? From [the `build` docs](https://pypa-build.readthedocs.io/en/stable/differences.html#setup-py-sdist-bdist-wheel):

> build is roughly the equivalent of `setup.py sdist bdist_wheel` but with [PEP 517](https://www.python.org/dev/peps/pep-0517) support, allowing use with projects that don’t use setuptools.

and [this SO post](https://stackoverflow.com/questions/17804185/what-does-setup-py-clean-exactly-do) I found looking for the purpose of `setup.py clean`:

> As far as I know, it just removes the the `build` subdirectory, where Python puts all the files to be installed, including extensions that need to be compiled.

I guess neither the `rm -rfv dist augur.egg-info` nor `python3 setup.py clean` are necessary when used in the context of this workflow which starts with a fresh clone of the repo, but maybe best to keep them in case we ever need to run this manually.

Reply:
My recollection is that `build` uses temporary directories to provide proper build isolation, thus there won't ever be any `build/` dir to remove even outside of the CI context. Maybe also applies to `augur.egg-info`? But `dist/` could be present outside of CI.

Reply:
In a quick test, `build/` doesn't appear when using `python -m build` (at least on a successful run) but `nextstrain_augur.egg-info` still does (and final artifacts are still in `dist/`). Note the existing script seems to misspell the egg-info name. Hmm, so I'd say fix the egg-info name, remove `setup.py clean`, and add `build/` to the `rm -rfv` invocation?

Reply:
Sounds good, done.